src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_factory.py

```diff
@@ -10,6 +10,12 @@ def iot_hub_service_factory(_):
     return get_mgmt_service_client(IotHubClient).iot_hub_resource
 
 
+def iot_hub_certificate_factory(_):
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.iothub.iot_hub_client import IotHubClient
+    return (get_mgmt_service_client(IotHubClient).iot_hub_resource, get_mgmt_service_client(IotHubClient).certificates)
```

Comment:
You can simply return `get_mgmt_service_client(IotHubClient)` and then based on what you need in your commands, reference `.iot_hub_resource` or `.certificates` as needed. It would make your custom commands much easier to read.

Reply:
Okay, let me investigate this!